client/beefy/src/worker.rs

```diff
@@ -508,7 +557,9 @@ where
 			},
 			RoundAction::Enqueue => {
 				debug!(target: "beefy", "🥩 Buffer justification for round: {:?}.", block_num);
-				self.pending_justifications.entry(block_num).or_insert(justification);
+				if self.pending_justifications.try_insert(block_num, justification).is_err() {
+					error!(target: "beefy", "🥩 Buffer justification dropped for round: {:?}.", block_num);
+				}
```

Comment:
Or you can use the original `BTreeMap` and just add a check inline right here:

```suggestion
				if self.pending_justifications.len() < MAX_BUFFERED_JUSTIFICATIONS {
				self.pending_justifications.entry(block_num).or_insert(justification);
				} else {
					error!(target: "beefy", "🥩 Buffer justification dropped for round: {:?}.", block_num);
				}
```

This is the single place where the justifs map is being inserted into.